spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitUtils.java

```diff
@@ -208,16 +195,25 @@ public static boolean isExclusiveUseChannelClose(ShutdownSignalException sig) {
 				&& ((AMQP.Channel.Close) shutdownReason).getReplyText().contains("exclusive");
 	}
 
-	public static Object determineShutdownReason(ShutdownSignalException sig) {
-		if (shutDownSignalReasonMethod == null) {
-			return false;
-		}
-		try {
-			return ReflectionUtils.invokeMethod(shutDownSignalReasonMethod, sig);
+	public static boolean isMismatchedQueueArgs(Exception e) {
```

Comment:
How about some JavaDocs for such a public API ?
Yes, I see that many methods here are without them, but we can address that on demand.
Right now I vote for this one :smile: 
I think something like log sample for the `AMQP.PRECONDITION_FAILED` in the JavaDoc would be useful, too.